### DISCUSSION TITLE
Sending messages _right now_

### DISCUSSION THREAD

**Discussion by kykyi** [excid3/noticed#446] [Q&A]
Hi @excid3 and all!

I am really enjoying working with the gem, thanks for all the hard work 💪 

I am trying to understand if I can create a notification and send it _right now_ by not going on the `ActiveJob` queue, but rather sending inline? Kind of akin to `SomeSidekiqJob.perform_sync` vs `SomeSidekiqJob.perform_async`.

Some code:
```
user.with_lock do
   user.must_be_true = false 

   UserNotifier.with(message: "Hello world").deliver(user, deliver_now: true)
   
   user.save! # raises an error
end
```
By wrapping in the transaction, if there was any error in the saving of the user, or the sending of the notification, then everything inside the transaction would not be committed to the db (ofc) but I would also not want the notification to be sent (or queued to be sent!)

From what I can see this behaviour does not currently exist, with the main indication of this being the `:deliver` method!
https://github.com/excid3/noticed/blob/1514ae3f239b991a3b952e3805474c147e68ea4e/app/models/concerns/noticed/deliverable.rb#L112-L113

If I set `enqueue_job: false`, then the the notification will not be enqueued and I will not have any notification send iiuc?

Ideally some sort of ` UserNotifier.with(message: "Hello world").deliver(user, deliver_now: true)` would be the API which could be used in this case. 

Happy to open a PR but this could be a deviation away from your direction with the gem. 

Many thanks!

**Comment by kykyi** [comment]
Digging a little more into the code I can see there is tests for `enqueue: true` on the email mailer, I presume you can then set `enqueue: false` but I'm not sure this is related to my query above. Fwiw I am sending sms and email 😄

**Comment by excid3** [comment]
Noticed v2 always enqueues jobs for sending. I don't know if we'll support inline executions because they will slow down your requests. If you don't set a delay, the emails will be sent basically immediately in your background workers. For example, Sidekiq might have your email sent before the HTTP request finishes.

You can use the `async` adapter with ActiveJob to execute all jobs inline.
